### PR TITLE
NodeGraph: Detect dataframes more accurately based on fields (#47213)

### PR DESCRIPTION
**Pull request by connorlindsey**
* NodeGraph: Detect dataframes more accurately based on fields

* Make get fields case insensitive

* Update node graph docs

(cherry picked from commit 939a7781113b0d2a0e5bd7b078891a9fd6695b85)

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by joey-grafana — APPROVED** [review]
LGTM 👍

File: docs/sources/visualizations/node-graph.md [hunk 1]
```diff
@@ -88,14 +88,14 @@ Required fields:
...
+| ------------- | ------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| title         | string        | Name of the node visible in just under the node.                                                                                                                                               |
+| subtitle      | string        | Additional, name, type or other identifier shown under the title.                                                                                                                              |
+| mainstat      | string/number | First stat shown inside the node itself. It can either be a string showing the value as is or a number. If it is a number, any unit associated with that field is also shown.                  |
+| secondarystat | string/number | Same as mainStat, but shown under it inside the node.                                                                                                                                          |
```
Comment (achatterjee-grafana) [inline docs/sources/visualizations/node-graph.md:96]:
```suggestion
| secondarystat | string/number | Same as `mainStat`, shown under it inside the node.                                                                                                                                          |
```

File: docs/sources/visualizations/node-graph.md [hunk 2]
```diff
@@ -88,14 +88,14 @@ Required fields:
...
+| arc\_\_\*     | number        | Any field prefixed with `arc__` will be used to create the color circle around the node. All values in these fields should add up to 1. You can specify color using `config.color.fixedColor`. |
```
Comment (achatterjee-grafana) [inline docs/sources/visualizations/node-graph.md:97]:
```suggestion
| arc\_\_\*     | number        | Any field prefixed with `arc__` to create the color circle around the node. All values in these fields should add up to one (1). You can specify color using `config.color.fixedColor`. |
```

File: docs/sources/visualizations/node-graph.md [hunk 3]
```diff
@@ -88,14 +88,14 @@ Required fields:
...
+| detail\_\_\*  | string/number | Any field prefixed with `detail__` will be shown in the header of context menu when clicked on the node. Use `config.displayName` for more human readable label.                               |
```
Comment (achatterjee-grafana) [inline docs/sources/visualizations/node-graph.md:98]:
```suggestion
| detail\_\_\*  | string/number | Any field prefixed with `detail__` will be shown in the header of the context menu when clicked on the node. Use `config.displayName` for more human-readable label.                               |
```

**Review by achatterjee-grafana — COMMENTED** [review]
I had a few doc comments. Leaving them here.

**Pull request merged**